app/javascript/packs/homePageFeed.jsx

```diff
@@ -43,6 +43,92 @@ function sendFeaturedArticleAnalyticsGA4(articleId) {
   })();
 }
 
+function feedConstruct(
+  pinnedItem,
+  imageItem,
+  feedItems,
+  bookmarkedFeedItems,
+  bookmarkClick,
+  currentUserId,
+  timeFrame,
+) {
+  const commonProps = {
+    bookmarkClick,
+  };
+
+  const feedStyle = JSON.parse(document.body.dataset.user).feed_style;
+
+  if (imageItem) {
+    sendFeaturedArticleGoogleAnalytics(imageItem.id);
+    sendFeaturedArticleAnalyticsGA4(imageItem.id);
+  }
+
+  return feedItems.map((item) => {
+    // billboard is an html string
+    if (typeof item === 'string') {
+      return (
+        <div
+          key={item.id}
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{
+            __html: item,
+          }}
+        />
+      );
+    }
+
+    if (Array.isArray(item) && item[0].podcast) {
+      return <PodcastEpisodes key={item.id} episodes={item} />;
+    }
+
+    if (typeof item === 'object') {
+      // For "saveable" props, "!=" is used instead of "!==" to compare user_id
+      // and currentUserId because currentUserId is a String while user_id is an Integer
+
+      if (item.id === pinnedItem?.id && timeFrame === '') {
+        return (
+          <Article
+            {...commonProps}
+            key={item.id}
+            article={pinnedItem}
+            pinned={true}
```

Comment:
I admit I'm having a bit of trouble following here, it's a little hard to tell what bits are _different in each return blob, some details `like `key` or `article` seem to be more-or-less the same (or, at least, _could_ be `item`) — in this blob, it looks like `pinned` might be the _only_ bit that's changed from the basic blob, maybe the `item.id === pinnedItem?.id` check could move here?

Reply:
I just copied this across from mostly what was previously done, but I can dive in a bit and refactor it out as well by seeing if there is commonalities.

Reply:
This was a REALLY good call @jaw6, it turns out that this can actually all be condensed into one object like below. 
```
if (typeof item === 'object') {
      return <Article
        {...commonProps}
        key={item.id}
        article={item}
        pinned={item.id === pinnedItem?.id}
        isFeatured={item.id === imageItem?.id}
        feedStyle={feedStyle}
        isBookmarked={bookmarkedFeedItems.has(item?.id)}
        saveable={item.user_id != currentUserId}
        //For "saveable" props, "!=" is used instead of "!==" to compare user_id
        // and currentUserId because currentUserId is a String while user_id is an Integer
      />
    }
```
Since Ben has approved already and I don't want to cause us to re-request a review and test again, I'll add this to a subsequent PR with some other improvements if thats okay - unless you find some blocking changes in your QA then Ill add it along with those changes because then we'd need re-test the PR anyways.

Reply:
I've updated this in https://github.com/forem/forem/pull/19600